src/api/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -476,7 +476,17 @@ public boolean fullfillRecipe(final LootContext context, final List<IItemHandler
                 {
                     if(citizen != null && ItemStackUtils.compareItemStackListIgnoreStackSize(tools, stack, false, true) && ItemStackUtils.getDurability(handler.getStackInSlot(slotOfStack)) > 0 )
                     {
-                        citizen.getInventoryCitizen().damageInventoryItem(slotOfStack, 1, citizen, item -> item.sendBreakAnimation(Hand.MAIN_HAND));
+                        ItemStack toDamage = handler.extractItem(slotOfStack,1, false);
```

Comment:
@someaddons didn't you add these days some utility methods for this?

Reply:
ye inventorycitizen has a shrink(slot) thing

Reply:
eh this is just literally the same as `citizen.getInventoryCitizen().damageInventoryItem`? :D why the change

Reply:
Because it's not a citizen inventory?

Reply:
but you cant like just craft with items in some storage

Reply:
The sifter has been crafting against storage (as much private crafting is) for quite some time now. There is no RS request, and thus no scope for how much to pick up.

Reply:
What is the main problem with sifting from the worker inv?

Reply:
Oh, even if we do move to crafter inventory, we still can't use the citizeninventory helper here without major restructuring of the code. The recipe gets called with a generic handler set, because private crafting goes against the building. (this is the bug I had to fix a few days ago with context) The only reason we even have access to the citizen is because of loottable/context, but we don't know if the inventory belongs to the citizen or the building.  The code I replaced in this PR works if everything is in the citizen, by accident of parallel objects being equivalent. When run against the building, it was attempting to damage an empty slot, since the handler didn't match the citizen inventory.